OpenRA.Mods.Common/Activities/Air/HeliLand.cs

```diff
@@ -19,26 +19,42 @@ public class HeliLand : Activity
 		readonly Aircraft aircraft;
 		readonly WDist landAltitude;
 		readonly bool requireSpace;
+		readonly Actor ignoreActor;
 
 		bool soundPlayed;
+		bool landingInitiated;
 
-		public HeliLand(Actor self, bool requireSpace)
-			: this(self, requireSpace, self.Info.TraitInfo<AircraftInfo>().LandAltitude) { }
+		public HeliLand(Actor self, bool requireSpace, Actor ignoreActor = null)
+			: this(self, requireSpace, self.Info.TraitInfo<AircraftInfo>().LandAltitude, ignoreActor) { }
 
-		public HeliLand(Actor self, bool requireSpace, WDist landAltitude)
+		public HeliLand(Actor self, bool requireSpace, WDist landAltitude, Actor ignoreActor = null)
 		{
 			this.requireSpace = requireSpace;
 			this.landAltitude = landAltitude;
+			this.ignoreActor = ignoreActor;
 			aircraft = self.Trait<Aircraft>();
 		}
 
 		public override Activity Tick(Actor self)
 		{
 			if (IsCanceling)
+			{
+				aircraft.RemoveInfluence();
 				return NextActivity;
+			}
 
-			if (requireSpace && !aircraft.CanLand(self.Location))
-				return this;
+			if (requireSpace && !landingInitiated)
+			{
+				var landingCell = self.Location;
+				if (!aircraft.CanLand(landingCell, ignoreActor))
+				{
+					self.NotifyBlocker(landingCell);
```

Comment:
Pinging the blocker every tick isn't great for performance. Please add a delay between retries like you do in `Land`.

Reply:
Done.